File: services/server/src/main/java/com/yaytsa/server/infrastructure/persistence/entity/ListeningSessionEntity.java
package com.yaytsa.server.infrastructure.persistence.entity;

import jakarta.persistence.*;
import java.time.OffsetDateTime;
import java.util.UUID;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "listening_session")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class ListeningSessionEntity {

  @Id
  @GeneratedValue(strategy = GenerationType.UUID)
  private UUID id;

  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(
      name = "user_id",
      nullable = false,
      foreignKey = @ForeignKey(name = "fk_listening_session_user"))
  private UserEntity user;

  @Column(columnDefinition = "jsonb", nullable = false)
  private String state = "{}";

  @Column(name = "started_at", nullable = false)
  private OffsetDateTime startedAt;

  @Column(name = "last_activity_at", nullable = false)
  private OffsetDateTime lastActivityAt;

  @Column(name = "ended_at")
  private OffsetDateTime endedAt;

  @Column(name = "session_summary", columnDefinition = "TEXT")
  private String sessionSummary;
}

File: yay-tsa-v2/infra-persistence/adaptive/src/main/kotlin/dev/yaytsa/persistence/adaptive/entity/ListeningSessionEntity.kt
package dev.yaytsa.persistence.adaptive.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table
import org.hibernate.annotations.JdbcTypeCode
import org.hibernate.type.SqlTypes
import java.time.Instant
import java.util.UUID

@Entity
@Table(name = "listening_sessions", schema = "core_v2_adaptive")
class ListeningSessionEntity(
    @Id
    val id: UUID = UUID.randomUUID(),
    @Column(name = "user_id", nullable = false)
    val userId: UUID = UUID.randomUUID(),
    @Column(name = "state", nullable = false, length = 20)
    val state: String = "ACTIVE",
    @Column(name = "started_at", nullable = false)
    val startedAt: Instant = Instant.now(),
    @Column(name = "last_activity_at", nullable = false)
    val lastActivityAt: Instant = Instant.now(),
    @Column(name = "ended_at")
    val endedAt: Instant? = null,
    @Column(name = "session_summary")
    val sessionSummary: String? = null,
    val energy: Float? = null,
    val intensity: Float? = null,
    @JdbcTypeCode(SqlTypes.ARRAY)
    @Column(name = "mood_tags", columnDefinition = "text[]")
    val moodTags: List<String> = emptyList(),
    @Column(name = "attention_mode", nullable = false)
    val attentionMode: String = "",
    @Column(name = "seed_track_id")
    val seedTrackId: UUID? = null,
    @JdbcTypeCode(SqlTypes.ARRAY)
    @Column(name = "seed_genres", columnDefinition = "text[]")
    val seedGenres: List<String> = emptyList(),
    @Column(name = "version", nullable = false)
    var version: Long = 0,
)
